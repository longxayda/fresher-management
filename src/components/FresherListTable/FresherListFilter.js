import React, { useState, useEffect } from "react";
import { Container, Row, Col } from "react-bootstrap";
import fresherService from "services/fresherManagement/fresherService";

function FresherListFilter({ setFilterParam }) {
  const [yearList, setYearList] = useState([]);
  const [courseList, setCourseList] = useState([]);
  const [filterError, setFilterError] = useState({});

  const [filterYear, setFilterYear] = useState(0);
  const [filterCourse, setFilterCourse] = useState("");

  useEffect(async () => {
    const fetchYear = async () => {
      const response = await fresherService.getYearList();
      if (response?.success === false) {
        return {
          error: true,
          probMessage: response.response.message,
          data: [],
        };
      }
      if (response.data) {
        return { data: response.data };
      } else {
        return { data: [] };
      }
    };

    try {
      const years = await fetchYear();
      let x = years.data.sort();
      x.reverse();
      setYearList(x);
    } catch {
      setYearList([]);
    }
    return () => {
      setYearList([]);
    };
  }, []);

  useEffect(async () => {
    if (filterYear !== 0) {
      try {
        const response = await fresherService.getCourseList(filterYear);
        if (response?.success === false) {
          setCourseList([]);
        } else {
          let x = response.data;
          setCourseList(x ?? []);
        }
      } catch {
        setCourseList([]);
      }
    } else {
      setCourseList([]);
    }
    return () => {
      setCourseList([]);
    };
  }, [filterYear]);

  function handleFilterEvent(e) {
    setFilterCourse(e.target.value);
    setFilterParam({
      year: filterYear,
      course: e.target.value,
    });
  }

  function handleChooseYear(e) {
    if (e.target.value == 0) {
      setFilterCourse("");
      setFilterParam({
        year: 0,
        course: "",
      });
    } else {
      setFilterYear(e.target.value);
    }
  }

  return (
    <>
      <Container fluid className="ps-3 pe-3 fresher-list-filter">
        <Row>
          <Col
            md="6"
            lg="5"
            xl="4"
            xxl="2"
            className="p-1 "
            style={{ width: 120 }}
          >
            <select
              className="form-select"
              aria-label="Default select example"
              onChange={(e) => handleChooseYear(e)}
              defaultValue={0}
            >
              <option value="0">All Year</option>
              {yearList.map((year, i) => {
                return (
                  <option key={i} value={year}>
                    {year}
                  </option>
                );
              })}
            </select>
          </Col>
          <Col
            md="6"
            lg="5"
            xl="4"
            xxl="2"
            className="p-1"
            style={{ width: 120 }}
          >
            <select
              className="form-select"
              aria-label="Default select example"
              onChange={(e) => handleFilterEvent(e)}
              defaultValue={filterCourse}
            >
              <option value="" hidden={filterYear != 0}>
                All Class
              </option>
              {courseList.map((course, i) => {
                return (
                  <option key={i} value={course.classId}>
                    {course.classCode}
                  </option>
                );
              })}
            </select>
          </Col>
        </Row>
      </Container>
    </>
  );
}

export default FresherListFilter;
