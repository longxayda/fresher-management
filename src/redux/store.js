import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
