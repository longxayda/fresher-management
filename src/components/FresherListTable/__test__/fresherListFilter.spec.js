import React from "react";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { mount } from "enzyme";
Enzyme.configure({ adapter: new Adapter() });
import store from "redux/store";
import { Provider } from "react-redux";
import FresherListFilter from "components/FresherListTable/fresherListFilter";

describe("Test FresherList FresherListFilter", () => {
  let component;
  beforeEach(() => {
    component = mount(
      <Provider store={store}>
        {" "}
        <FresherListFilter />{" "}
      </Provider>
    );
  });

  test("has the correct class", () => {
    expect(component.find(".fresher-list-filter").exists()).toEqual(true);
  });

  test("render FresherListFilter's component correctly", () => {
    component = mount(
      <Provider store={store}>
        {" "}
        <FresherListFilter />{" "}
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });

  test("has select dropdown", () => {
    expect(component.find("select").exists()).toEqual(true);
  });

  // describe("choosing an option", () => {
  //   beforeEach(() => {
  //     component
  //       .find("select")
  //       .first()
  //       .simulate("change", { target: { value: "2022" } });
  //   });

  //   test("shows that text in the select area", () => {
  //     expect(component.find("select").first().prop("value")).toEqual("2022");
  //   });
  // });
});
