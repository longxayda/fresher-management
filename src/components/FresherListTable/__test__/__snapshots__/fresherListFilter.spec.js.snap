// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FresherList FresherListFilter render FresherListFilter's component correctly 1`] = `" <div class=\\"ps-3 pe-3 fresher-list-filter container-fluid\\"><div class=\\"row\\"><div style=\\"width: 120px;\\" class=\\"p-1  col-xxl-2 col-xl-4 col-lg-5 col-md-6\\"><select class=\\"form-select\\" aria-label=\\"Default select example\\"><option value=\\"0\\" selected=\\"\\">All Year</option></select></div><div style=\\"width: 120px;\\" class=\\"p-1 col-xxl-2 col-xl-4 col-lg-5 col-md-6\\"><select class=\\"form-select\\" aria-label=\\"Default select example\\"><option value=\\"\\" selected=\\"\\">All Class</option></select></div></div></div> "`;
